arviz/data/io_pyjags.py

```diff
@@ -0,0 +1,326 @@
+            self.pyjags = pyjags
+        except ModuleNotFoundError:
+            self.pyjags = None
+
+    def _pyjags_samples_to_xarray(self,
+                                  pyjags_samples: tp.Dict[str, np.ndarray]) \
+            -> tp.Tuple[xarray.Dataset, xarray.Dataset]:
+        data, data_warmup = get_draws(pyjags_samples=pyjags_samples,
+                                      warmup_iterations=self.warmup_iterations,
+                                      warmup=self.save_warmup)
+
+        return (
+            dict_to_dataset(data,
+                            library=self.pyjags,
+                            coords=self.coords,
+                            dims=self.dims),
+            dict_to_dataset(data_warmup,
+                            library=self.pyjags,
+                            coords=self.coords,
+                            dims=self.dims),
+        )
+
+    @requires("posterior")
+    def posterior_to_xarray(self) -> tp.Tuple[xarray.Dataset, xarray.Dataset]:
+        """Extract posterior samples from fit."""
+        return self._pyjags_samples_to_xarray(self.posterior)
+
+    @requires("prior")
+    def prior_to_xarray(self) -> tp.Tuple[xarray.Dataset, xarray.Dataset]:
+        """Extract posterior samples from fit."""
+        return self._pyjags_samples_to_xarray(self.prior)
+
+    def to_inference_data(self):
+        """Convert all available data to an InferenceData object."""
+        # obs_const_dict = self.observed_and_constant_data_to_xarray()
+        # predictions_const_data = self.predictions_constant_data_to_xarray()
+        return InferenceData(
+            **{
+                "posterior": self.posterior_to_xarray(),
+                "prior": self.prior_to_xarray(),
+            }
+        )
+
+
+def get_draws(pyjags_samples: tp.Dict[str, np.ndarray],
+              variables: tp.Optional[tp.Union[str, tp.Iterable[str]]] = None,
+              warmup: bool = False,
+              warmup_iterations: int = 0) \
+        -> tp.Tuple[tp.Dict[str, np.ndarray], tp.Dict[str, np.ndarray]]:
+    """
+
+    Parameters
+    ----------
+    pyjags_samples: a didctionary mapping variable names to NumPy arrays of MCMC
+                    chains of samples with shape
+                    (parameter_dimension, chain_length, number_of_chains)
+
+    variables: the variables to extract from the samples dictionary
+    warmup: whether or not to return warmup draws in data_warmup
+    warmup_iterations: the number of warmup iterations if any
+
+    Returns
+    -------
+
+    """
+    data_warmup = OrderedDict()
+
+    if variables is None:
+        variables = list(pyjags_samples.keys())
+    elif isinstance(variables, str):
+        variables = [variables]
+
+    if not isinstance(variables, Sequence):
+        raise TypeError('variables must be of type Sequence or str')
+
+    variables = tuple(variables)
+
+    if warmup_iterations > 0:
+        warmup_samples, actual_samples =\
```

Comment:
You can just put everything to oneline and then let black handle multiline version.

Reply:
I tried black but don't like the way it formats things. Among other things, it often goes above 80 characters per line. Is using black mandatory for ArviZ?

Reply:
Yes. And we use 100 char per line.

Reply:
I ran black with 100 characters per line which produces lines that are longer than 100. This causes pylint to complain:
arviz/tests/external_tests/test_data_pyjags.py:110:0: C0301: Line too long (138/100) (line-too-long)
arviz/tests/external_tests/test_data_pyjags.py:123:0: C0301: Line too long (130/100) (line-too-long)
arviz/tests/external_tests/test_data_pyjags.py:137:0: C0301: Line too long (130/100) (line-too-long)
arviz/tests/external_tests/test_data_pyjags.py:141:0: C0301: Line too long (115/100) (line-too-long)

If I run black with 80 or 90 characters per line it also produces lines that are longer than 100.

If I manually insert line breaks to meet the 100 character limit then black complains that the file is not formatted the way it would format it. Is there a way to either disable the pylint check for characters per line or the black check for individual files?